### ISSUE TITLE
Fails to `go install` with Go 1.17+

### ISSUE THREAD

**Issue by rwxrob** [kubernetes/kompose#1472]
Fails to install using the standard installation method for all packages
under Go 1.17 and greater. The alternatives (`curl`) prevent this from
being a blocker, but this is a significant bug at the moment.

> :warning
> Leaving `replace` in a `go.mod` file will force a failure of `go
install` as of Go 1.17. The Kubernetes `kompose project` is current
failing because of this

```
$ go install github.com/kubernetes/kompose@latest
go install: github.com/kubernetes/kompose@latest (in github.com/kubernetes/kompose@v1.26.1):
	The go.mod file for the module providing named packages contains one
  or more replace directives. It must not contain directives that 
  would cause it to be interpreted differently than if it were the 
  main module.
```

**Comment by cdrage** [comment]
Hi @rwxrob unfortunately this is intentional. 

We use an older (deprecated) library for libcompose as well as numerous other modules from both Docker, OpenShift and Kubernetes that use the older `github.com/Sirupsen/logrus` library, which was replaced with `github.com/sirupsen/logrus` (notice the lowercase). 

All so we can support older versions of docker-compose as well as compatibility with Kubernetes and OpenShift.

We do the same with numerous other packages in order to maintain equilibrium between Docker, Kubernetes and OpenShift. All three of which use different versions of different modules. After endless hours of trying to figure out which libraries work well with each other, we managed to at least create `go.mod` and `go.sum` thanks to the help of @hangyan 

The best analogy I can give is when you fix 1 bug, and 99 others appear. We've tried to fix one module, and then Kubernetes breaks. Try to fix Kubernetes and then libcompose compatibility breaks. Try to fix that, and then we have to use the new libcompose library. Try to use the new libcompose library and then we're back to fixing Kubernetes and OpenShift modules because they are outdated and not compatible.

Sorry for the long explanation / rant, but had to get it out there so people in the future realize why we can't implement `go install` and have to refer to users to use the binary installation method.
